### DISCUSSION TITLE
Does uasyncio not have a queue class/object?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [micropython/micropython#9141] [General]
This is take from a recently nightly build...

>>> import uasyncio
>>> dir(uasyncio)
['__class__', '__getattr__', '__name__', '__path__', 'CancelledError', 'Task', 'TaskQueue', 'select', 'sleep', 'sleep_ms', 'sys', 'ticks_add', 'ticks_diff', 'core', 'wait_for', 'gather', 'Event', 'Lock', '__version__', '_attrs', 'wait_for_ms', 'ThreadSafeFlag', 'open_connection', 'start_server', 'StreamReader', 'StreamWriter', 'TimeoutError', 'SingletonGenerator', 'IOQueue', 'Loop', 'create_task', 'run_until_complete', 'run', 'get_event_loop', 'current_task', 'new_event_loop', 'ticks']
>>> uasyncio.__version__
(3, 0, 0)

I thought I read somewhere uasyncio 3.0 had a (generic) queue class/object?  Creating a simple array/list queue class is pretty easy to do, but would not want to recreate the wheel.

**Comment by peterhinch** [comment]
There is no official class. An unofficial one, with the same API as CPython's `Queue` class, is described [here](https://github.com/peterhinch/micropython-async/blob/master/v3/docs/TUTORIAL.md#35-queue).

**Comment by Jibun-no-Kage** [comment]
Ah, I saw the unofficial one, cool, I wrote one that is much simpler since all I needed was a queue to drive flashing an RGB diode with specific sequence or pattern without overlap.  Maybe mine is closer to a FIFO ring buffer now that I think about it.  But, the unofficial one is a very interesting implementation.  Thanks.
